internal/proxy/rate_limit_interceptor.go

```diff
@@ -37,7 +37,7 @@ func RateLimitInterceptor(limiter types.Limiter) grpc.UnaryServerInterceptor {
 		if err == nil {
 			limit, rate := limiter.Limit(rt, n)
```

Comment:
The API is a little redundant
maybe limiter can just return err here?
err := limiter.check(rt, n)
if err !=nil {
   return err
}

Reply:
`limiter.check` needs to return results to distinguish if the request is limited or denied. like:
```Golang
isLimited, isDenied := limiter.check(rt, n)
if isDenied {
    return deniedResponse
}
if isLimited {
    return limitedResponse
}
```